### PR TITLE
infoschema: add metric_summary table to query all summary metrics.

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Add metric_summary table to query all summary metrics. This table can quickly help User to find abnormal metrics between two different times.

eg:

```sql
-- query all metric in specified time range.
>select * from `METRIC_SUMMARY` where time > "2020-01-16 16:00:54.933" and time < "2020-01-16 16:20:54.933" order by `SUM_VALUE` desc;
+--------------------------------------------------+----------------------------+----------------------------+-----------------------------+--------------------------+---------------------------+
| METRIC_NAME                                      | TIME                       | SUM_VALUE                  | AVG_VALUE                   | MIN_VALUE                | MAX_VALUE                 |
+--------------------------------------------------+----------------------------+----------------------------+-----------------------------+--------------------------+---------------------------+
| pd_scheduler_store_status                        | 2020-01-16 16:00:54.935000 |          6.07010244119e+13 | 37938140257.5               |        0.0               |         4.99963170816e+11 |
| pd_cluster_status                                | 2020-01-16 16:00:54.935000 |          4.99993793161e+13 |           2.27269905982e+11 |        0.0               |         2.49981585408e+12 |
| heap_mem_usage                                   | 2020-01-16 16:00:54.935000 | 4599581104.0               |    76659685.0667            | 13458600.0               | 144483352.0               |
| pd_scheduler_config                              | 2020-01-16 16:00:54.935000 |    4043868.0               |       10641.7578947         |        0.0               |    200000.0               |
| pd_schedule_filter                               | 2020-01-16 16:00:54.935000 |    2177780.39715           |        5731.00104513        |        0.0               |     16819.2               |
| pd_checker_event_count                           | 2020-01-16 16:00:54.935000 |     983775.596047          |        6148.59747529        |        0.0               |     12340.8               |
| pd_balance_scheduler_status                      | 2020-01-16 16:00:54.935000 |     854400.000068          |        2848.00000023        |        0.0               |     18000.3600072         |
| pd_server_etcd_state                             | 2020-01-16 16:00:54.935000 |     136190.0               |        2269.83333333        |        2.0               |      3614.0               |
| pd_scheduler_balance_leader                      | 2020-01-16 16:00:54.935000 |      24000.0000019         |          75.000000006       |        0.0               |       120.002400048       |
| tidb_gc_config                                   | 2020-01-16 16:00:54.935000 |      24000.0               |         600.0               |      600.0               |       600.0               |
| pd_region_syncer_status                          | 2020-01-16 16:00:54.935000 |      16220.0               |         270.333333333       |        0.0               |       411.0               |
| goroutines_count                                 | 2020-01-16 16:00:54.935000 |      13374.0               |         222.9               |       58.0               |       494.0               |
| qps_ideal                                        | 2020-01-16 16:00:54.935000 |      10659.8168425         |         532.990842126       |      272.970938769       |       658.660271834       |
| kv_write_size_0.999                              | 2020-01-16 16:00:54.935000 |       2812.672             |         140.6336            |      127.872             |       255.488             |
...
...
...
-- compare 2 different time range metrics and order by ratio.
>select t1.sum_value /t2.sum_value as ratio, t1.*,t2.* from 
  metric_summary as t1 join metric_summary as t2 
where
  t1.metric_name = t2.metric_name and  
  t1.time > "2020-01-16 16:00:54.933" and t1.time < "2020-01-16 16:10:54.933" and  
  t2.time > "2020-01-16 16:10:54.933" and t2.time < "2020-01-16 16:20:54.933" 
  order by ratio desc;
+----------------+--------------------------------------------------+----------------------------+----------------------------+-----------------------------+--------------------------+---------------------------+--------------------------------------------------+----------------------------+----------------------------+-----------------------------+--------------------------+---------------------------+
| ratio          | METRIC_NAME                                      | TIME                       | SUM_VALUE                  | AVG_VALUE                   | MIN_VALUE                | MAX_VALUE                 | METRIC_NAME                                      | TIME                       | SUM_VALUE                  | AVG_VALUE                   | MIN_VALUE                | MAX_VALUE                 |
+----------------+--------------------------------------------------+----------------------------+----------------------------+-----------------------------+--------------------------+---------------------------+--------------------------------------------------+----------------------------+----------------------------+-----------------------------+--------------------------+---------------------------+
| 3.98099801726  | etcd_wal_fsync_duration_0.999                    | 2020-01-16 16:00:54.935000 |          0.76241939798     |           0.076241939798    |        0.0159746666667   |         0.507648          | etcd_wal_fsync_duration_0.999                    | 2020-01-16 16:10:54.935000 |          0.191514638961    |           0.0191514638961   |        0.0159805714286   |         0.031904          |
| 3.80554485331  | etcd_wal_fsync_duration_0.99                     | 2020-01-16 16:00:54.935000 |          0.712193979798    |           0.0712193979798   |        0.0157466666667   |         0.46848           | etcd_wal_fsync_duration_0.99                     | 2020-01-16 16:10:54.935000 |          0.18714638961     |           0.018714638961    |        0.0158057142857   |         0.03104           |
| 1.70232615487  | parse_duration_0.999                             | 2020-01-16 16:00:54.935000 |          0.00581689417069  |           0.000581689417069 |        0.000159655652174 |         0.00241664        | parse_duration_0.999                             | 2020-01-16 16:10:54.935000 |          0.00341702684532  |           0.000341702684532 |        0.0001596832      |         0.00062416        |
| 1.58604950456  | compile_duration_0.999                           | 2020-01-16 16:00:54.935000 |          0.0426435966904   |           0.00426435966904  |        0.00063736        |         0.0097536         | compile_duration_0.999                           | 2020-01-16 16:10:54.935000 |          0.0268866744498   |           0.00268866744498  |        0.000636741818182 |         0.00973312        |
| 1.2205518808   | pd_client_cmd_duration_0.9                       | 2020-01-16 16:00:54.935000 |          0.036912249279    |           0.00111855300845  |        0.00045           |         0.0152            | pd_client_cmd_duration_0.9                       | 2020-01-16 16:10:54.935000 |          0.0302422616028   |           0.000916432169783 |        0.00045           |         0.0076            |
| 1.21517165667  | pd_client_cmd_duration_0.8                       | 2020-01-16 16:00:54.935000 |          0.0314422818744   |           0.000952796420435 |        0.0004            |         0.0144            | pd_client_cmd_duration_0.8                       | 2020-01-16 16:10:54.935000 |          0.0258747656776   |           0.000784083808412 |        0.0004            |         0.0072            |
| 1.16797177117  | distsql_execution_duration_0.999                 | 2020-01-16 16:00:54.935000 |          0.156865116139    |           0.0156865116139   |        0.00399307692308  |         0.03128           | distsql_execution_duration_0.999                 | 2020-01-16 16:10:54.935000 |          0.134305571429    |           0.0134305571429   |        0.00399357142857  |         0.06328           |
| 1.16745022159  | tidb_cop_duration_0.999                          | 2020-01-16 16:00:54.935000 |          0.156865116139    |           0.0156865116139   |        0.00399307692308  |         0.03128           | tidb_cop_duration_0.999                          | 2020-01-16 16:10:54.935000 |          0.134365571429    |           0.0134365571429   |        0.00399357142857  |         0.06328           |
| 1.150920075    | etcd_wal_fsync_duration_0.9                      | 2020-01-16 16:00:54.935000 |          0.177468686869    |           0.0177468686869   |        0.0134666666667   |         0.0352            | etcd_wal_fsync_duration_0.9                      | 2020-01-16 16:10:54.935000 |          0.154197229437    |           0.0154197229437   |        0.0140571428571   |         0.0224            |
| 1.13597971803  | compile_duration_0.99                            | 2020-01-16 16:00:54.935000 |          0.0175244688645   |           0.00175244688645  |        0.0006136         |         0.005376          | compile_duration_0.99                            | 2020-01-16 16:10:54.935000 |          0.0154267444976   |           0.00154267444976  |        0.000607418181818 |         0.0051712         |
| 1.08059627511  | distsql_execution_duration_0.99                  | 2020-01-16 16:00:54.935000 |          0.128651161388    |           0.0128651161388   |        0.00393076923077  |         0.0248            | distsql_execution_duration_0.99                  | 2020-01-16 16:10:54.935000 |          0.119055714286    |           0.0119055714286   |        0.00393571428571  |         0.0568            |
| 1.07517774772  | tidb_cop_duration_0.99                           | 2020-01-16 16:00:54.935000 |          0.128651161388    |           0.0128651161388   |        0.00393076923077  |         0.0248            | tidb_cop_duration_0.99                           | 2020-01-16 16:10:54.935000 |          0.119655714286    |           0.0119655714286   |        0.00393571428571  |         0.0568            |
| 1.05729231864  | parse_duration_0.99                              | 2020-01-16 16:00:54.935000 |          0.00263294170692  |           0.000263294170692 |        0.000156556521739 |         0.0006464         | parse_duration_0.99                              | 2020-01-16 16:10:54.935000 |          0.0024902684532   |           0.00024902684532  |        0.000156832       |         0.0004816         |
| 1.04348646467  | load_schema_ops                                  | 2020-01-16 16:00:54.935000 |          0.47999960012     |           0.023999980006    |        0.0               |         0.060001200024    | load_schema_ops                                  | 2020-01-16 16:10:54.935000 |          0.459996000304    |           0.0229998000152   |        0.0               |         0.06              |
| 1.042526244    | pd_grpc_completed_commands_duration_0.8          | 2020-01-16 16:00:54.935000 |          0.310166666667    |           0.00424885844749  |        0.004             |         0.022             | pd_grpc_completed_commands_duration_0.8          | 2020-01-16 16:10:54.935000 |          0.297514492754    |           0.00407554099662  |        0.004             |         0.009             |
| 1.0407349867   | pd_grpc_completed_commands_duration_0.9          | 2020-01-16 16:00:54.935000 |          0.3476875         |           0.00476284246575  |        0.0045            |         0.0235            | pd_grpc_completed_commands_duration_0.9          | 2020-01-16 16:10:54.935000 |          0.334078804348    |           0.00457642197737  |        0.0045            |         0.0095            |
| 1.02876387364  | query_duration_0.9                               | 2020-01-16 16:00:54.935000 |          0.0714184896476   |           0.00714184896476  |        0.00542857142857  |         0.0078            | query_duration_0.9                               | 2020-01-16 16:10:54.935000 |          0.0694216539652   |           0.00694216539652  |        0.0065            |         0.00764705882353  |
| 1.02746618532  | pd_scheduler_region_heartbeat                    | 2020-01-16 16:00:54.935000 |        224.399256025       |           0.897597024102    |        0.0               |        10.8               | pd_scheduler_region_heartbeat                    | 2020-01-16 16:10:54.935000 |        218.400624012       |           0.87360249605     |        0.0               |        13.2               |
| 1.02423649562  | qps_ideal                                        | 2020-01-16 16:00:54.935000 |       5393.72423646        |         539.372423646       |      466.344551745       |       658.660271834       | qps_ideal                                        | 2020-01-16 16:10:54.935000 |       5266.09260607        |         526.609260607       |      272.970938769       |       595.209594103       |
| 1.01268997302  | pd_client_cmd_duration_0.99                      | 2020-01-16 16:00:54.935000 |          0.0469550537352   |           0.00142288041622  |        0.000495          |         0.01592           | pd_client_cmd_duration_0.99                      | 2020-01-16 16:10:54.935000 |          0.0463666620448   |           0.001405050365    |        0.000495          |         0.00796           |
| 1.01211730991  | pd_handle_transactions_rate                      | 2020-01-16 16:00:54.935000 |          3.33999360038     |           0.333999360038    |        0.319993600128    |         0.36              | pd_handle_transactions_rate                      | 2020-01-16 16:10:54.935000 |          3.30000640013     |           0.330000640013    |        0.32              |         0.36              |
| 1.01181788224  | tidb_cop_duration_0.9                            | 2020-01-16 16:00:54.935000 |          0.0372473107775   |           0.00372473107775  |        0.00330769230769  |         0.00391666666667  | tidb_cop_duration_0.9                            | 2020-01-16 16:10:54.935000 |          0.0368122677323   |           0.00368122677323  |        0.00335714285714  |         0.0041            |
| 1.01100358616  | distsql_execution_duration_0.9                   | 2020-01-16 16:00:54.935000 |          0.0373073840375   |           0.00373073840375  |        0.00330769230769  |         0.00391666666667  | distsql_execution_duration_0.9                   | 2020-01-16 16:10:54.935000 |          0.0369013369963   |           0.00369013369963  |        0.00335714285714  |         0.0041            |
| 1.00936300186  | go_gc_cpu_usage                                  | 2020-01-16 16:00:54.935000 |          0.00196395458919  |           6.54651529731e-05 |       -4.82018280819e-06 |         0.000175346847331 | go_gc_cpu_usage                                  | 2020-01-16 16:10:54.935000 |          0.00194573665329  |           6.48578884429e-05 |       -4.84387043513e-06 |         0.000173825324795 |
...
...
...


```

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: executor/metric_reader.go [hunk 1]
```diff
@@ -193,3 +194,53 @@ func (c *queryClient) URL(ep string, args map[string]string) *url.URL {
...
+		return nil, nil
+	}
+	e.retrieved = true
+	totalRows := make([][]types.Datum, 0, len(infoschema.MetricTableMap))
+	e.quantiles = []float64{0.999, 0.99, 0.90, 0.80}
+	tps := make([]*types.FieldType, 0, len(e.table.Columns))
+	for _, col := range e.table.Columns {
+		tps = append(tps, &col.FieldType)
+	}
+	startTime := e.extractor.StartTime.Format("2006-01-02 15:04:05.999")
```
Comment (lonng) [inline executor/metric_reader.go]: We have used this format in many places, prefer to define a constant for all metrics related time column.

Comment (crazycs520) [reply executor/metric_reader.go]: great, done.

File: executor/metric_reader.go [hunk 2]
```diff
@@ -193,3 +194,53 @@ func (c *queryClient) URL(ep string, args map[string]string) *url.URL {
...
+	retrieved bool
+}
+
+func (e *MetricSummaryRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) ([][]types.Datum, error) {
+	if e.retrieved || e.extractor.SkipRequest {
+		return nil, nil
+	}
+	e.retrieved = true
+	totalRows := make([][]types.Datum, 0, len(infoschema.MetricTableMap))
+	e.quantiles = []float64{0.999, 0.99, 0.90, 0.80}
```
Comment (lonng) [inline executor/metric_reader.go]:
```suggestion
	e.quantiles = []float64{1, 0.999, 0.99, 0.90, 0.80}
```

File: executor/metric_reader.go [hunk 3]
```diff
@@ -193,3 +194,53 @@ func (c *queryClient) URL(ep string, args map[string]string) *url.URL {
...
+				totalRows = append(totalRows, row.GetDatumRow(tps))
+			}
+		}
+	}
+	return totalRows, nil
+}
+
+func (e *MetricSummaryRetriever) genMetricQuerySQLS(name, startTime, endTime string, quantile float64) []string {
+	if quantile == 0 {
+		sql := fmt.Sprintf(`select "%s",min(time),sum(value),avg(value),min(value),max(value) from metric_schema.%s where time > '%s' and time < '%s'`, name, name, startTime, endTime)
```
Comment (lonng) [inline executor/metric_reader.go]: Does the sum value make sense?

Comment (crazycs520) [reply executor/metric_reader.go]: sometime it is useful, Such as `AWR` has `SUM CPU TIME`

File: executor/metric_reader.go [hunk 4]
```diff
@@ -193,3 +194,53 @@ func (c *queryClient) URL(ep string, args map[string]string) *url.URL {
...
+	retrieved bool
+}
+
+func (e *MetricSummaryRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) ([][]types.Datum, error) {
+	if e.retrieved || e.extractor.SkipRequest {
+		return nil, nil
+	}
+	e.retrieved = true
+	totalRows := make([][]types.Datum, 0, len(infoschema.MetricTableMap))
+	e.quantiles = []float64{1, 0.999, 0.99, 0.90, 0.80}
```
Comment (lonng) [inline executor/metric_reader.go]: `quantiles` should be a local variable instead of field.

File: executor/metric_reader.go [hunk 5]
```diff
@@ -193,3 +194,53 @@ func (c *queryClient) URL(ep string, args map[string]string) *url.URL {
...
+	tps := make([]*types.FieldType, 0, len(e.table.Columns))
+	for _, col := range e.table.Columns {
+		tps = append(tps, &col.FieldType)
+	}
+	startTime := e.extractor.StartTime.Format(plannercore.MetricTableTimeFormat)
+	endTime := e.extractor.EndTime.Format(plannercore.MetricTableTimeFormat)
+	for name, def := range infoschema.MetricTableMap {
+		sqls := e.genMetricQuerySQLS(name, startTime, endTime, def.Quantile)
+		for _, sql := range sqls {
+			rows, _, err := sctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(sql)
```
Comment (djshow832) [inline executor/metric_reader.go]: According to your manual test, does it need to retry after network failure?

File: infoschema/tables.go [hunk 6]
```diff
@@ -106,6 +106,8 @@ const (
 	tableTiFlashReplica    = "TIFLASH_REPLICA"
 	// TableInspectionResult is the string constant of inspection result table
 	TableInspectionResult = "INSPECTION_RESULT"
+	// TableMetricSummary is a summary table that contain all metrics.
```
Comment (djshow832) [inline infoschema/tables.go]:
```suggestion
	// TableMetricSummary is a summary table that contains all metrics.
```

File: executor/metric_reader.go (hunk 4 above)
Comment (crazycs520) [reply executor/metric_reader.go]: done.

File: executor/metric_reader.go (hunk 5 above)
Comment (crazycs520) [reply executor/metric_reader.go]: When querying the single metric tables, there is already has the retry internal.

**Pull request merged**